extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/ServerSideEncryptingAmazonS3.java

```diff
@@ -43,9 +51,10 @@
 
 /**
  * {@link AmazonS3} wrapper with {@link ServerSideEncryption}. Every {@link AmazonS3#putObject},
- * {@link AmazonS3#copyObject}, {@link AmazonS3#getObject}, and {@link AmazonS3#getObjectMetadata} methods should be
+ * {@link AmazonS3#copyObject}, {@link AmazonS3#getObject}, and {@link AmazonS3#getObjectMetadata},
+ * {@link AmazonS3#initiateMultipartUpload}, @{@link AmazonS3#uploadPart} methods should be
  * wrapped using ServerSideEncryption.
- *
+ * <p>
```

Comment:
Note for future work: if encryption is needed, and there is only one right way to do it, then it should be done automatically by an `EncryptedS3StorageConnector` rather than forcing clients to add the code. Forcing clients to do that means clients have to be aware of the target file system, which is what this great new API tries to avoid.

Reply:
Yup that makes sense :)